ui/library/src/context/ScrollContext.ts

```diff
@@ -1,19 +1,59 @@
 import * as React from 'react';
 
+import { NodeDestination } from '../components/types/outline';
 import { Nullable } from '../components/types/utils';
 import { logProviderWarning } from '../utils/provider';
+import { generatePageIdFromIndex } from '../utils/scroll';
 import ScrollDetector, { ScrollDirection } from '../utils/ScrollDirectionDetector';
+import VisibleEntriesDetector from '../utils/VisibilityEntriesDetector';
+
+export type PageInfo = {
+  pageNumber?: number;
+  pageIndex?: number;
+};
```

Comment:
`Info` sounds too general for me.... I wonder if you can use `PageNumber` directly and put comments telling that `pageNumber` starts from 1 while `pageIndex` starts from 0.

Reply:
@yensung alright i will do that